src/query/sql/src/planner/plans/join.rs

```diff
@@ -587,5 +602,83 @@ fn update_statistic(
     if let Some(new_ndv) = new_stat.ndv {
         left_col_stat.ndv = new_ndv;
         right_col_stat.ndv = new_ndv;
+        for (idx, left) in left_statistics.column_stats.iter_mut() {
+            if idx == &left_index {
+                continue;
+            }
+            // Other columns' histograms are inaccurate, so make them None
+            left.histogram = None;
+        }
+        for (idx, right) in right_statistics.column_stats.iter_mut() {
+            if idx == &right_index {
+                continue;
+            }
+            right.histogram = None;
+        }
+    }
+}
+
+// Prune the buckets that are not in the range of [new_min, new_max]
+fn prune_buckets(
+    left_hist: &Histogram,
+    right_hist: &Histogram,
+    new_min: &Option<Datum>,
+    new_max: &Option<Datum>,
+) -> Result<(Vec<HistogramBucket>, Vec<HistogramBucket>)> {
+    if let Some(new_min) = new_min && let Some(new_max) = new_max {
+        let mut left_buckets = Vec::new();
+        let mut right_buckets = Vec::new();
+        for (idx, bucket) in left_hist.buckets.iter().enumerate() {
+            if idx == 0 {
+                continue;
+            }
+            if left_hist.buckets[idx-1].upper_bound() <= new_max && bucket.upper_bound() >= new_min {
```

Comment:
Maybe it should be `left_hist.buckets[idx-1].upper_bound() >= new_min && bucket.upper_bound() <= new_max`.

Reply:
As long as there is an intersection, we won't prune the bucket.

Reply:
Yes, but the condition in the code will always be false.

Reply:
I think it should be `(left_hist.buckets[idx-1].upper_bound() >= new_min && left_hist.buckets[idx-1].upper_bound() <= new_max) || (bucket.upper_bound() > new_min && bucket.upper_bound() < new_max)`.

Reply:
I think it should be `left_hist.buckets[idx-1].upper_bound() <= new_max && bucket.upper_bound() > new_min`.

Reply:
I think it should be `left_hist.buckets[idx-1].upper_bound() <= new_max && bucket.upper_bound() > new_min`.